sim/team-validator.ts

```diff
@@ -1613,6 +1613,19 @@ export class TeamValidator {
 		if (species.baseSpecies === "Greninja" && toID(set.ability) === 'battlebond') {
 			set.species = "Greninja-Bond";
 		}
+
+		if (species.baseSpecies === "Unown" && dex.gen === 2) {
+			let resultBinary = '';
+			for (const iv of ['atk', 'def', 'spe', 'spa'] as const) {
+				resultBinary += Math.floor(set.ivs[iv] / 2).toString(2).padStart(4, '0').slice(1, 3);
+			}
+			const resultDecimal = Math.floor(parseInt(resultBinary, 2) / 10);
+			const expectedLetter = String.fromCharCode(resultDecimal + 65);
+			const unownLetter = species.name === "Unown" ? "A" : species.name.charAt(6);
```

Comment:
I think this might work:
```suggestion
			const unownLetter = species.forme || "A";
```